astropy/table/table.py

```diff
@@ -856,9 +856,17 @@ def pprint(self, max_lines=None, max_width=None, show_name=True,
             else:
                 print(line)
 
+    def _make_index_row_display_table(self, index_row_name='idx'):
+        if index_row_name not in self.columns:
```

Comment:
One thing you might do here is keep on adding a space to `index_row_name` until it isn't in the original table.  Since this is just a display column, I think the "hidden" spaces in the name wouldn't really be a problem.

In fact you could take this to the extreme and have the default `index_row_name` be `' '` (which is a perfectly legal column name).  Then it would end up looking like pandas notebook output where the row index column does not have a name label.